packages/devtools_server/lib/src/server.dart

```diff
@@ -203,6 +212,15 @@ Future<void> registerLaunchDevToolsService(
 
     service.registerServiceCallback(launchDevToolsService, (params) async {
       try {
+        // First see if we have an existing DevTools client open that we can
+        // reuse.
+        final reusableClient = await clients.findReusableClient();
+        if (reusableClient != null) {
+          await reusableClient.connectToVmService(vmServiceUri);
+          // reusableClient.bringToFront() / Notify / whatever...
```

Comment:
the bringToFront part is the million £ question. I expect notify is the best we will be able to do.

Reply:
I went with notifications, though I have a note to do some testing of how `window.focus()` behaves these days, in case that works well.